packages/typescript/internal/ts_project.bzl

```diff
@@ -48,13 +48,19 @@ def _join(*elements):
 def _ts_project_impl(ctx):
     arguments = ctx.actions.args()
 
-    generated_srcs = False
+    has_generated = False
```

Comment:
ooh, this looks like a nice bugfix, if helpful I can repro in a standalone test case for ts_project and land as a prefactoring

Reply:
There's 2 things here:
* if there are no source files after 1+ generated files in `ctx.files.srcs` then the `fail` would never be hit, this was a bug
* the `--rootDir` was only absolute if using generated sources. I honestly don't recall why I did this or if it correct, but I assume I did it try to to fix/workaround https://github.com/angular/angular/issues/36290 - but maybe that is only an angular bug and not ts_project bug?

If you'd like to take this commit and add tests then go for it :+1:

Reply:
I've updated that commit to output a better error messages showing the last files found in the generated vs src folders, for example the error this PR is currently getting which I commented about below:
```
srcs cannot be a mix of generated files and source files: bazel-out/k8-fastbuild/bin/src/app/todos/todos.component.css.map, src/app/todos/todos.component.html
```